### PR TITLE
Not allow invalid amounts

### LINKED ISSUE #580: [auction details] place order button becomes enabled if amount=0
1. open app
2. follow Auctions from menu
3. open auction 
4. click MAX or enter 0
5. fill in the price with 0 as well

**Result:** Place order button becomes available 
 
**Exp.result:**  Place order button is disabled. 
Also, it would be nice to show validation messages for the fields when they are empty/filled in with 0 amounts

### PR DESCRIPTION
**Pull request by henrypalacios**
closes #580 

When an amount is placed, [a parse attempt is made](https://github.com/gnosis/ido-ux/blob/8accd6ba7b49d5a21e06859c0a5672da3890202c/src/state/orderPlacement/hooks.ts#L171-L186) on the bidding token.

**Proposal**: Invalidate amount when it could not be parsed.

- This will invalidate the 0.
- This would invalidate amounts exceeding the bidding token's decimal places. (I don't know why anyone would want to do that) related #593 

> :bulb: I don't think it is necessary but the error in case of decimals could be better specified.

### CODE REVIEW

**Review by hpmaxi — APPROVED** [review]

**Comment by elena-zh** [comment]
Hi @henrypalacios , could you please provide a review link to test these changes?
Thanks

**Comment by elena-zh** [comment]
@henrypalacios , #580 issue LGTM! However, I'd rather rewrite validation message for the Price field in a manner 'Price must me higher than 0', as such messages are shown when validation price on a min value.
higher image

Case #231 (#593 duplicates it) is not fixed in this PR: 
image

Thanks!

**Comment by elena-zh** [comment]
Hey @henrypalacios , great! Now everything looks good to me (#580 and #231 cases).
The only 'curious' thing is that there is no validation for an inverted price field on number of decimals. 
However, I can successfully place an order with an inverted price with 20 decimals, and it works fine. Hence, there is no need to fix it and this issue can be closed. 

Btw, could you please take a look into the old #262 issue if you have a spare time? It is also related to the price field.

Thanks

File: src/state/orderPlacement/hooks.ts [hunk 1]
```diff
@@ -216,32 +216,47 @@ export const useGetOrderPlacementError = (
...
   const insufficientBalance =
     balanceIn &&
     amountIn &&
     balanceIn.lessThan(amountIn) &&
     `Insufficient ${getTokenDisplay(amountIn.token, chainId)}` + ' balance.'
 
   const priceEqualsZero =
-    amountIn && price && (price === '0' || price === 'Infinity') ? `Price must not be 0` : undefined
+    amountIn && price && (price === '0' || price === 'Infinity' || price === '.' || price === '0.')
+      ? `Price must be higher than 0`
```
Comment (josojo) [inline src/state/orderPlacement/hooks.ts:252]: This is also not optimal, is it? 

Firstt the user gets the message: "higher than 0", and then next he gets the message "higher than x", where x is the current auction clearing price

Comment (henrypalacios) [reply src/state/orderPlacement/hooks.ts:252]: :bulb:  Let me unify the message.

**Comment by elena-zh** [comment]
Hi @henrypalacios , changes for ordinary price LGTM!
However, 
1. when a price is inverted, it is a bit confusing to see than amount should be less than <min price> when we enter 0. 
image
It would be nice to save the logic for 'inverted prices' 
 - 0: 'Price must be higher than 0'
 - and then, if an amount is greater than min price, show 'Price must be lower than <min price>'

2. Then, the next case is related to an auction state when order cancellation is not allowed. Min price is used for validation
image


User should see message 'price should be higher than <current price>'  in such case

image

**Comment by elena-zh** [comment]
@henrypalacios , I think it might help you

```
ordinary prices + orders are allowed to be cancelled
		price = 0 :  'price should be higher than <min bidding price>' 
		price < min sell price: price should be higher than <min sell price>' 
		
		
ordinary prices + orders are NOT allowed to be cancelled		
		price = 0 :  'price should be higher than <current price>' 
		price < min sell price: price should be higher than <current price>' 
		price > min sell price and < current price: price should be higher than <current price>'


inverted prices + orders are allowed to be cancelled
		price = 0 :  'price should be higher than 0' 
		price > max sell price: price should be lower than <max sell price>' 

inverted prices + orders are NOT allowed to be cancelled
		price = 0 :  'price should be higher than 0' 
		price > max sell price: price should be lower than <current price>' 	
		price < max sell price and > current price:	price should be lower than <current price>' 
```

File: src/state/orderPlacement/hooks.ts [hunk 2]
```diff
@@ -220,28 +225,37 @@ export const useGetOrderPlacementError = (
...
+  const messageInitialPrice = () =>
+    showPricesInverted ? `Price must be higher than 0` : messageHigherClearingPrice
+
   const priceEqualsZero =
-    amountIn && price && (price === '0' || price === 'Infinity') ? `Price must not be 0` : undefined
+    amountIn && price && (price === '0' || price === 'Infinity' || price === '.' || price === '0.')
+      ? messageInitialPrice()
+      : undefined
+  const invalidSellAmount =
+    sellAmount && amountIn && price && !sellAmountScaled && `Invalid Bidding price`
```
Comment (josojo) [inline src/state/orderPlacement/hooks.ts:258]: nit: bidding price

**Review by josojo — APPROVED** [review]
LGTM

**Comment by elena-zh** [comment]
Hi @henrypalacios , 
cool, thank you! 
I have found the last case that remains to be unfixed: 
```
ordinary prices + orders are allowed to be cancelled
		price = 0 :  'price should be higher than <min bidding price>'
```

Currently, it shows validation message 'price should be higher than <current price>' instead of the <min bidding price>
https://drive.google.com/file/d/1xSB7isUV8l2d0P1eDBuB3Q65scpiX8F7/view
Could you please fix it?

All the rest cases look good to me now. 

Thanks

**Comment by elena-zh** [comment]
@henrypalacios , cool! Now all changes look good to me!

**Pull request merged**